src/Uno.Templates/content/unoapp/MyExtensionsApp._1/Services/Caching/WeatherCache.cs

```diff
@@ -38,7 +38,7 @@ public async ValueTask<IImmutableList<WeatherForecast>> GetForecast(Cancellation
 #if (useLogging)
             _logger.LogWarning("App is offline and cannot connect to the API.");
 #endif
-            throw new Exception("No internet connection");
+            throw new InvalidOperationException("No internet connection");
```

Comment:
Thinking maybe change this to:
```suggestion
            throw new NetworkException(NetworkExceptionCategory.Offline, 0, "No internet connection");
```